package/kedro_viz/server.py

```diff
@@ -59,6 +67,43 @@ def _hash(value):
     return hashlib.sha1(value.encode("UTF-8")).hexdigest()[:8]
 
 
+def _check_viz_up(port):
+    url = "http://127.0.0.1:{}/".format(port)
+    response = requests.get(url)
+    return response.status_code == 200
+
+
+# pylint: disable=unused-argument
+def run_viz(port=None, line=None) -> None:
+    """
+    Line magic function to start kedro viz. It calls a kedro viz in a thread and display it in
+    the Jupyter notebook environment.
+
+    Args:
+        port: TCP port that viz will listen to. Defaults to 4141.
+        line: line required by line magic interface.
+
+    """
+    if not port:  # Default argument doesn't work in Jupyter line magic
+        port = 4141
+
+    if port not in _VIZ_THREADS:
```

Comment:
maybe you should close the old viz and start a new one if the port is already in use

Reply:
Updated, and we are using multiprocessing instead of threading now.